Core/include/Acts/Propagator/EigenStepper.ipp

```diff
@@ -15,49 +17,26 @@ auto Acts::EigenStepper<B, E, A>::boundState(State& state,
                                              const Surface& surface,
                                              bool reinitialize) const
     -> BoundState {
-  // Transport the covariance to here
-  std::optional<Covariance> covOpt = std::nullopt;
-  if (state.covTransport) {
-    covarianceTransport(state, surface, reinitialize);
-    covOpt = state.cov;
-  }
-  // Create the bound parameters
-  BoundParameters parameters(state.geoContext, std::move(covOpt), state.pos,
-                             state.p * state.dir, state.q, state.t,
-                             surface.getSharedPtr());
-  // Create the bound state
-  BoundState bState{std::move(parameters), state.jacobian,
-                    state.pathAccumulated};
-  // Reset the jacobian to identity
-  if (reinitialize) {
-    state.jacobian = Jacobian::Identity();
-  }
-  /// Return the State
-  return bState;
+  FreeVector parameters;
+  parameters << state.pos[0], state.pos[1], state.pos[2], state.t, state.dir[0],
+      state.dir[1], state.dir[2], state.q / state.p;
+  return detail::boundState(state.geoContext, state.cov, state.jacobian,
+                            state.jacTransport, state.derivative,
+                            state.jacToGlobal, parameters, state.covTransport,
+                            state.pathAccumulated, surface, reinitialize);
```

Comment:
I like this, it will make the Stepper code more readable and it is actually clear what is happening.